help_function/src/onedpl_test_group_load.cpp

```diff
@@ -0,0 +1,106 @@
+// ====------ onedpl_test_group_load.cpp---------- -*- C++ -* ----===////
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+// ===----------------------------------------------------------------------===//
+#include <sycl/sycl.hpp>
+#include <dpct/dpct.hpp>
+#include <dpct/dpl_utils.hpp>
+#include <iostream>
+
+
+bool helper_function(const int* ptr,const char* func_name){
+  int expected[512];
+  for (int i = 0; i < 512; ++i) {
+      expected[i] = i;
+  }
+  for (int i = 0; i < 512; ++i) {
+    if (ptr[i] != expected[i]) {
+      std::cout << func_name <<" failed\n";
+      std::ostream_iterator<int> Iter(std::cout, ", ");
+      std::copy(ptr, ptr + 512, Iter);
+      std::cout << std::endl;
+      return false;
+    }
+  }
+
+  std::cout << func_name <<" pass\n";
+  return true;
+}
+
+bool test_load_blocked() {
+  sycl::queue q;
+  int data[512];
+  for (int i = 0; i < 512; i++) data[i] = i;
+
+  sycl::buffer<int, 1> buffer(data, 512);
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d = dacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          dpct::group::load_blocked<128>(item.get_local_linear_id(), d, thread_data);
+        });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_only);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_function(ptr,"test_load_blocked");
+}
+
+bool test_load_striped() {

```

Comment:
It looks like `test_load_striped` is running the exact same check as `test_load_blocked` which only checks if the input data was unaltered.  Don't we need some check that the threads loaded the appropriate values?

I would suggest having an "output" buffer as well for which teach thread writes to a known contiguous region the elements it loaded from the input.  Then you can check that this output is as you would expect it when you are back on the host (striped with a specific stride or contiguous depending on the type of load).

Right now, as far as I can see, there is no test that this functions as expected, only that it doesn't have inappropriate side effects.

Reply:
Yes correct, we will need to change this.

Reply:
Made changes to the interface for checking, could you please review.